### ISSUE TITLE
TS: End time isn't updated after entering in format HH.mm

### ISSUE THREAD

**Issue by Dravere** [kimai/kimai#695]
#### Expected behaviour
If you enter the start time in the time sheet in the format HH.mm.ss or HH.mm (e.g. 10.00) it will change the format after losing focus to HH:mm:ss. Which is good because nothing else is accepted when submitting the form. This should work for the end time as well.

#### Actual behaviour
But it doesn't work for the end time. When the input field loses focus/a change is registered the entered time string stays in its format (e.g. 10.00) instead of reformatting to the accepted form HH:mm.

#### Steps to reproduce the behaviour
Add a new time to the timesheet. Select a project, an activity, enter a description, leave the date as is, enter for the start time 10.00 and for the end time 11.00, click on submit.

The reason the point notation is handy: You can use your numblock on the keyboard.

#### Kimai version
Just checked with the latest master.

#### How to solve it
In the file ./extensions/ki_timesheets/js/ts_func.js
Line 475, function ts_durationToTime
I added the new function ts_formatTime and ts_formatDate and changed ts_durationToTime to use them and also to put the value for \#end_time and \#end_day which wasn't done before.
```javascript
function ts_formatTime(value) {
    var H = value.getHours();
    var i = value.getMinutes();
    var s = value.getSeconds();

    if (H<10) H = "0"+H;
    if (i<10) i = "0"+i;
    if (s<10) s = "0"+s;
    
    return H + ":" + i + ":" + s;
}

function ts_formatDate(value) {
    var d = value.getDate();
    var m = value.getMonth() + 1;
    var y = value.getFullYear();
    
    if (d<10) d = "0"+d;
    if (m<10) m = "0"+m;
    
    return d + "." + m + "." + y;
}

function ts_durationToTime() {
    end = ts_getEndDate();
    durationArray=$("#duration").val().split(/:|\./);
    if(end!=null && durationArray.length > 0 && durationArray.length < 4) {
        secs = durationArray[0]*3600;
        if(durationArray.length > 1)
            secs += (durationArray[1]*60);
        if(durationArray.length > 2)
            secs += parseInt(durationArray[2]);
        begin = new Date();
        begin.setTime(end.getTime()-(secs*1000));

        $("#start_time").val(ts_formatTime(begin));
        $("#end_time").val(ts_formatTime(end));
        $("#start_day").val(ts_formatDate(begin));
        $("#end_day").val(ts_formatDate(end));
    }
}
```

**Comment by simonschaufi** [comment]
Thank you for your contribution but could you please make a pull request with your changes? That would help us a lot. Please use develop branch for it.

**Comment by Dravere** [comment]
I will try. Never done this before. But I'm willing to learn and try it. Just give me a couple days :)

**Comment by simonschaufi** [comment]
Resolved in #699
